packages/flutter_tools/gradle/src/main/kotlin/dependency_version_checker.gradle.kts

```diff
@@ -0,0 +1,270 @@
+        // to match the current values of the "warn" versions in the next release.
+        // Before updating any "error" version, ensure that you have updated the corresponding
+        // "warn" version for a full release to provide advanced warning. See
+        // flutter.dev/go/android-dependency-versions for more.
+        val warnGradleVersion : Version = Version(7,0,2)
+        val errorGradleVersion : Version = Version(0,0,0)
+
+        val warnJavaVersion : JavaVersion = JavaVersion.VERSION_11
+        val errorJavaVersion : JavaVersion = JavaVersion.VERSION_1_1
+
+        val warnAGPVersion : Version = Version(7,0,0)
+        val errorAGPVersion : Version = Version(0,0,0)
+
+        val warnKGPVersion : Version = Version(1,5,0)
+        val errorKGPVersion : Version = Version(0,0,0)
+
+        /**
+         * Checks if the project's Android build time dependencies are each within the respective
+         * version range that we support. When we can't find a version for a given dependency
+         * we treat it as within the range for the purpose of this check.
+         */
+        fun checkDependencyVersions(project : Project) {
+            var agpVersion : Version? = null
+            var kgpVersion : Version? = null
+            Map map = new HashMap()
+
+            checkGradleVersion(getGradleVersion(project), project)
+            checkJavaVersion(getJavaVersion(project), project)
+            agpVersion = getAGPVersion(project)
+            if (agpVersion != null) {
+                checkAGPVersion(agpVersion, project)
+            } else {
+                project.logger.error("Warning: unable to detect project AGP version. Skipping " +
+                        "version checking. " + ignored)
+            }
+
+            kgpVersion = getKGPVersion(project)
+            if (kgpVersion != null) {
+                checkKGPVersion(kgpVersion, project)
+            } else {
+                project.logger.error("Warning: unable to detect project KGP version. Skipping " +
+                        "version checking.")
+            }
+        }
+
+        // https://docs.gradle.org/current/kotlin-dsl/gradle/org.gradle.api.invocation/-gradle/index.html#-837060600%2FFunctions%2F-1793262594
+        fun getGradleVersion(project : Project) : Version {
+            return Version.fromString(project.gradle.getGradleVersion())
+        }
+
+        // https://docs.gradle.org/current/kotlin-dsl/gradle/org.gradle.api/-java-version/index.html#-1790786897%2FFunctions%2F-1793262594
+        fun getJavaVersion(project : Project) : JavaVersion {
+            return JavaVersion.current()
+        }
+
+        // This approach is taken from AGP's own version checking plugin:
+        // https://android.googlesource.com/platform/tools/base/+/studio-master-dev/build-system/gradle-core/src/main/java/com/android/build/gradle/internal/utils/agpVersionChecker.kt#58.
```

Comment:
I think it'd be better to link to a concrete commit (I assume `studio-master-dev` is a branch). Otherwise the link might become broken

Reply:
Changed!